### PR TITLE
Platforms versions support

### LINKED ISSUE #569: [Feature] Support for 'platform subversions'
Is your feature request related to a problem? Please describe:
The naomi-2 slug is a 'version' of the 'arcade' platform.  as in /arcade/version/naomi-2.  RomM is currently unable to handle this ''subversion' of arcade?

Describe the solution you'd like:
RomM should support platform 'subversions' so that users can separate rom platform subtypes

image

### LINKED ISSUE #609: [Feature] ENV to define config.yml location
**Describe the solution you'd like**
I use named docker volumes and is not possible to mount a file when using NFS/SMB, is possible to create a ENV VAR to change the location of config.yml to allow put in any other random folder?

### PR DESCRIPTION
**Pull request by zurdi15**
This PR adds support for platform versions. For example: ``NAOMI`` as version of ``Arcade``. The association must be configured in the ``config.yml`` file in the new ``system.versions`` section.

Roms in the platform version will be scanned search using in the ``main platform id``  as IGDB doesn't provide games for api calls for platform versions.

Now ``config.yml`` file must be placed inside a ``config`` folder and bind it as follows to docker: ``/path/config:/romm/config`` and if the ``config.yml`` file doesn't exists it will be created empty automatically.

config

versions

Closes #569 
Closes #609 

WIP:

 - [x] Unit tests

### CODE REVIEW

File: examples/docker-compose.example.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ services:
     volumes:
       - "/path/to/library:/romm/library"
       - "/path/to/resources:/romm/resources" # [Optional] Path where roms metadata (covers) are stored
-      - "/path/to/config.yml:/romm/config.yml" # [Optional] Path where config is stored
+      - "/path/to/config:/romm/config" # [Optional] Path where config is stored
```
Comment (gantoine) [inline examples/docker-compose.example.yml:44]: we'll need to mark this as a breaking change in 3.0 release

Comment (gantoine) [inline examples/docker-compose.example.yml:44]: we'll need to mark this as a breaking change in 3.0 release

**Review by gantoine — APPROVED** [review]
great stuff! i think doing it this way via the config, as opposed to relying on IGDB's relationships, is the smarter idea long term. good call 👍🏼

File: examples/docker-compose.example.yml (hunk 1 above)
Comment (zurdi15) [reply examples/docker-compose.example.yml:44]: Yes absolutely!

**Pull request merged**